packages/visual-editor/src/components/CustomCodeSection.tsx

```diff
@@ -90,14 +125,28 @@ const CustomCodeSectionWrapper = ({
   javascript,
   puck,
 }: WithId<WithPuckProps<CustomCodeSectionProps>>) => {
-  if (!html) {
-    return puck.isEditing ? <EmptyCustomCodeSection /> : null;
-  }
+  const streamDocument = useDocument();
+
+  const stableStreamDoc = React.useMemo(
+    () => streamDocument,
+    [JSON.stringify(streamDocument)]
+  );
 
   const containerRef = React.useRef<HTMLDivElement>(null);
   const scriptIdRef = React.useRef<number>(Math.floor(Math.random() * 1e9));
   const scriptTagId = `custom-code-section-script-${scriptIdRef.current}`;
 
+  const processedHtml = React.useMemo(
+    () => processHandlebarsTemplate(html, stableStreamDoc),
+    [html, stableStreamDoc]
+  );
+
+  // Process Handlebars in JavaScript string
+  const processedJavascript = React.useMemo(
+    () => processHandlebarsTemplate(javascript, stableStreamDoc),
```

Comment:
Handlebars isn't a js syntax so I don't think we want to use it here. But we do need to allow users to add embedded fields to the js. Ideally it's the same bracket notation as in our RTF fields.